### PR TITLE
Mlpmixer

### PR DESCRIPTION
**Pull request by soma2000-lang**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py [hunk 1]
```diff
@@ -290,3 +292,18 @@ def test_augment_boxes_ragged(self):
...
+        image = tf.random.uniform(shape=image_shape)
+
+        layer = preprocessing_utils.RandomCropAndResize(
+            target_size=self.target_size,
+            aspect_ratio_factor=(3 / 4, 4 / 3),
+            crop_area_factor=(0.8, 1.0),
+        )
+
+        input_resized = tf.image.resize(image, self.target_size)
+        output = layer(image, training=False)
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_crop_and_resize_test.py:307]: can we invoke this with training=True as well and check that all of the outputs are different?

Comment (soma2000-lang) [reply keras_cv/layers/preprocessing/random_crop_and_resize_test.py:307]: Sure

File: keras_cv/layers/preprocessing/random_crop_and_resize.py [hunk 2]
```diff
@@ -93,37 +92,38 @@ def __init__(
         self.bounding_box_format = bounding_box_format
         self.force_output_dense_images = True
 
-    def get_random_transformation(
-        self, image=None, label=None, bounding_box=None, **kwargs
+    def get_random_transformation_batch(
+        self, batch_size, label=None, bounding_box=None, **kwargs
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_crop_and_resize.py:96]: if not used, removed `label` and `bounding_box` arguments

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py [hunk 3]
```diff
@@ -290,3 +292,33 @@ def test_augment_boxes_ragged(self):
...
+        layer = preprocessing_utils.RandomCropAndResize(
+            target_size=self.target_size,
+            aspect_ratio_factor=(3 / 4, 4 / 3),
+            crop_area_factor=(0.8, 1.0),
+        )
+
+        input_resized = tf.image.resize(image, self.target_size)
+        output = layer(image, training=True)
+
+        self.assertNotAllClose(output, input_resized)
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_crop_and_resize_test.py:324]: lets instead do:

```
self.assertNotAllClose(output[0], output[1])
```

**Comment by LukeWood** [comment]
@soma2000-lang looks like the unit tests are failing!

**Comment by soma2000-lang** [comment]
@LukeWood looking into this.

**Comment by soma2000-lang** [comment]
@LukeWood  I am trying to fix the pip issue with my system.Till I fix this.

I just run the test using Google Colab and it seems to pass.

Link -https://colab.research.google.com/drive/1-TZpbQDMp1fe3mrfvouWAqH_PISf9SBh?usp=sharing

**Comment by LukeWood** [comment]
Screen Shot 2023-02-25 at 1 46 13 PM

**Comment by LukeWood** [comment]
@soma2000-lang left a comment including a merge conflict.  Can you resolve?

**Comment by soma2000-lang** [comment]
@LukeWood Resolved!

**Comment by soma2000-lang** [comment]
@LukeWood Resolved

**Review by LukeWood — CHANGES_REQUESTED** [review]
almost ready!  Please address remaining comments!

File: benchmarks/vectorized_crop_resize.py [hunk 4]
```diff
@@ -98,7 +98,7 @@ def __init__(
         self.force_output_dense_images = True
 
     def get_random_transformation(
-        self, image=None, label=None, bounding_box=None, **kwargs
+        self, image=None,**kwargs
```
Comment (LukeWood) [inline benchmarks/vectorized_crop_resize.py:101]: looks like image is also unused!

Comment (LukeWood) [inline benchmarks/vectorized_crop_resize.py:101]: be sure to run ./shell/format.sh before requesting pull request reviews

**Comment by soma2000-lang** [comment]
image
@LukeWood I had runned  ./shell/format.sh ,not sure though why the code format is still failing

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py [hunk 5]
```diff
@@ -290,3 +291,33 @@ def test_augment_boxes_ragged(self):
...
+
+        layer = preprocessing_utils.RandomCropAndResize(
+            target_size=self.target_size,
+            aspect_ratio_factor=(3 / 4, 4 / 5),
+            crop_area_factor=(0.8, 1.0),
+        )
+
+        input_resized = tf.image.resize(image, self.target_size)
+        output = layer(image, training=True)
+        output[1] = input_resized
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_crop_and_resize_test.py:322]: does this pass without this line?

**Review by LukeWood — CHANGES_REQUESTED** [review]
by adding:
```

        input_resized = tf.image.resize(image, self.target_size)
        output = layer(image, training=True)
        output[1] = input_resized
```

you're no longer testing that the images are independently augmented.  Please remove the assignment - if it fails it means the layer is not independently augmenting images.

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py [hunk 6]
```diff
@@ -290,3 +291,33 @@ def test_augment_boxes_ragged(self):
...
+            aspect_ratio_factor=(3 / 4, 4 / 5),
+            crop_area_factor=(0.8, 1.0),
+        )
+
+        input_resized = tf.image.resize(image, self.target_size)
+        output = layer(image, training=False)
+
+        self.assertAllClose(output, input_resized)
+
+    def random_crop_and_resize_on_batched_image_training(self):
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_crop_and_resize_test.py:310]: add test_ prefix or the test wont run

**Comment by LukeWood** [comment]
@soma2000-lang the goal is to independently augment each image.  I think getting this to work will require you to update the logic a bit.

**Comment by soma2000-lang** [comment]
@LukeWood I am working on it.Sorry for  the delay though.

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py [hunk 7]
```diff
@@ -116,7 +116,7 @@ def test_aspect_ratio_factor_errors(self, aspect_ratio_factor):
     def test_crop_area_factor_errors(self, crop_area_factor):
         with self.assertRaisesRegex(
             ValueError,
-            "`crop_area_factor` must be tuple of two positive floats less than or "
+            "`crop_area_factor` must be tuple of two positive floats less than or"  # noqa
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_crop_and_resize_test.py:119]: noqa ?

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py [hunk 8]
```diff
@@ -188,12 +188,12 @@ def test_augment_one_hot_segmentation_mask(self):
         self.assertAllClose(output["segmentation_masks"], input_mask_resized)
 
     def test_augment_bounding_box_single(self):
-        image = tf.zeros([20, 20, 3])
+        images = tf.zeros([20, 20, 3])
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_crop_and_resize_test.py:191]: this is actually a single image

**Comment by LukeWood** [comment]
It looks like this is a legitimate failure - can you fix the test failures?  @soma2000-lang

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py (hunk 7 above)
Comment (soma2000-lang) [reply keras_cv/layers/preprocessing/random_crop_and_resize_test.py:119]: Actually earlier the length of line were exceeding the limit ,will remove them now

File: keras_cv/layers/preprocessing/random_crop_and_resize_test.py (hunk 8 above)
Comment (soma2000-lang) [reply keras_cv/layers/preprocessing/random_crop_and_resize_test.py:191]: Yeah sorry  ,will do the necessary

**Comment by soma2000-lang** [comment]
@LukeWood I sent you an email 2 days back .I would be really greatfull if you can check ot once.


Regards

**Comment by soma2000-lang** [comment]
@james77777778  It would be great if youvcan help me with this.I am stuck on this for long.

**Comment by james77777778** [comment]
Hi @soma2000-lang 
I have roughly read your PR and I can share the idea about `self.get_random_transformation_batch`

https://colab.research.google.com/drive/15ed5Fqn5zrQClmmuvB1DWrDGwMrHA4Ka?usp=sharing

I think the vectorizing could be complete by modifying `self._transform_bounding_boxes` and `self._crop_and_resize` based on batched and independently sampled `transformations`

**Comment by soma2000-lang** [comment]
@james77777778 Thanks!

**Comment by LukeWood** [comment]
@soma2000-lang can you add a test like the one in https://github.com/keras-team/keras-cv/pull/1480/files#diff-1c771135874332a25b437da4a5706faea85ff719331dfe7beb255403a1a66477

**Comment by soma2000-lang** [comment]
@LukeWood sure!.Also sorry for the delay I will try to finish this pr soon.

**Comment by divyashreepathihalli** [comment]
@soma2000-lang can you please rebase with master?
